site/content/book/v2/helloworldpy.md

```diff
@@ -0,0 +1,170 @@
+from didcomm.secrets.secrets_resolver_demo import SecretsResolverDemo
+from didcomm.unpack import unpack, UnpackResult
+from didcomm.common.resolvers import ResolversConfig
+from didcomm.pack_encrypted import pack_encrypted, PackEncryptedConfig, PackEncryptedResult
+from didcomm.secrets.secrets_util import generate_x25519_keys_as_jwk_dict, generate_ed25519_keys_as_jwk_dict, jwk_to_secret
+from peerdid import peer_did
+from peerdid.did_doc import DIDDocPeerDID
+from peerdid.types import VerificationMaterialAuthentication, VerificationMethodTypeAuthentication, VerificationMaterialAgreement, VerificationMethodTypeAgreement, VerificationMaterialFormatPeerDID
+import json
+```
+### Step 3: Helpers
+In this step we add three helper functionalities:
+- **Secret resolver**
+This sample code needs a storage to keep the generated key pair secrets. It will then be referenced by the library as a `secrets_resolver`. We can instantiate it as follows:
+`secrets_resolver = SecretsResolverDemo()`
+Note that the `SecretsResolverDemo` simply stores the keys in a text file named `secrets.json`. As you've just realized, this secret storage is anything but secure. Keep in mind that securing keys is of utmost importance for a self-sovereign identity; never use it in production.
+- **DID Resolver**
+The following class is needed by the library to resolve the DID Peer into a DID Document and put it in the format required by the library:
+```
+class DIDResolverPeerDID(DIDResolver):
+    async def resolve(self, did: DID) -> DIDDoc:
+        did_doc_json = peer_did.resolve_peer_did(did, format = VerificationMaterialFormatPeerDID.JWK)
+        did_doc = DIDDocPeerDID.from_json(did_doc_json)
+
+        return DIDDoc(
+            did=did_doc.did,
+            key_agreement_kids = did_doc.agreement_kids,
+            authentication_kids = did_doc.auth_kids,
+            verification_methods = [
+                VerificationMethod(
+                    id = m.id,
+                    type = VerificationMethodType.JSON_WEB_KEY_2020,
+                    controller = m.controller,
+                    verification_material = VerificationMaterial(
+                        format = VerificationMaterialFormat.JWK,
+                        value = json.dumps(m.ver_material.value)
+                    )
+                )
+                for m in did_doc.authentication + did_doc.key_agreement
+            ],
+            didcomm_services = [
+                DIDCommService(
+                    id = s.id,
+                    service_endpoint = s.service_endpoint,
+                    routing_keys = s.routing_keys,
+                    accept = s.accept
+                )
+                for s in did_doc.service
+                if isinstance(s, DIDCommServicePeerDID)
+            ] if did_doc.service else []
+        )
+```
+- **DID Peer creation**
+We add the following function to simplify key generation, key storage, and the creation of a DID Peer:
```

Comment:
This can also be in the external .py file, so the main doc the reader deals with has less early clutter.

Reply:
same as last comment